pkg/volume/util/operationexecutor/operation_executor.go

```diff
@@ -330,6 +335,13 @@ type operationExecutor struct {
 	// pendingOperations keeps track of pending attach and detach operations so
 	// multiple operations are not started on the same volume
 	pendingOperations nestedpendingoperations.NestedPendingOperations
+
+	// recorder is used to record events in the API server
+	recorder record.EventRecorder
+
+	// represents a volume that should be attached to this node and
+	// mounted to the PodName.
+	volumeToMount VolumeToMount
```

Comment:
I don't think that's what you want, since `volumeToMount` is never initialized, it will essentially be an empty struct wherever it is used.

For some of these cases (detach) you simply don't have a pod object you can associate the event with.